File: lcof2/剑指 Offer II 036. 后缀表达式/Solution.go
func evalRPN(tokens []string) int {
	// https://github.com/emirpasic/gods#arraystack
	stack := arraystack.New()
	for _, token := range tokens {
		if len(token) > 1 || token[0] >= '0' && token[0] <= '9' {
			num, _ := strconv.Atoi(token)
			stack.Push(num)
		} else {
			y := popInt(stack)
			x := popInt(stack)
			switch token {
			case "+":
				stack.Push(x + y)
			case "-":
				stack.Push(x - y)
			case "*":
				stack.Push(x * y)
			default:
				stack.Push(x / y)
			}
		}
	}
	return popInt(stack)
}

func popInt(stack *arraystack.Stack) int {
	v, _ := stack.Pop()
	return v.(int)
}

File: lcof2/剑指 Offer II 036. 后缀表达式/Solution.swift
class Solution {
    func evalRPN(_ tokens: [String]) -> Int {
        var stk = [Int]()

        for token in tokens {
            if let num = Int(token) {
                stk.append(num)
            } else {
                let y = stk.removeLast()
                let x = stk.removeLast()
                switch token {
                case "+":
                    stk.append(x + y)
                case "-":
                    stk.append(x - y)
                case "*":
                    stk.append(x * y)
                default:
                    stk.append(x / y)
                }
            }
        }

        return stk.removeLast()
    }
}
